src/app/(frontend)/[id]/page.tsx

```diff
@@ -0,0 +1,92 @@
+"use client"
+import { useParams } from 'next/navigation';
+import React, { useState, useEffect } from 'react';
+import { Product } from '@/lib/models/ProductModel';
+import Image from 'next/image';
+import { register } from 'swiper/element/bundle';
+
+register();
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'swiper-container': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & {
+        navigation?: string;
+        pagination?: string;
+        loop?: string;
+      }, HTMLElement>,
+      'swiper-slide': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>,
+    }
+  }
+}
+
+const ProductDetailPage: React.FC = () => {
+  const [product, setProduct] = useState<Product | null>(null)
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const { id } = useParams()
+
+  useEffect(() => {
+    if (typeof id === 'string') {
+      setLoading(true);
+      fetch(`/api/${id}`)
+        .then((res) => {
+          if (res.ok) {
+            return res.json();
+          } else {
+            throw new Error('Failed to fetch product');
+          }
+        })
+        .then((data) => {
+          setProduct(data);
+          setLoading(false);
```

Comment:
React re-renders the page on each state update, this means that we're performing two renders on this place. We can avoid this by including `isLoading` in the `product` object.

```suggestion
          setProduct({ ...data, isLoading: false});
```